src/webrtc/call.ts

```diff
@@ -409,27 +387,41 @@ export class MatrixCall extends EventEmitter {
      * The audio will *not* be rendered from the remoteVideoElement.
      * @param {Element} element The <code>&lt;video&gt;</code> DOM element.
      */
-    setRemoteAudioElement(element : HTMLAudioElement) {
+    async setRemoteAudioElement(element : HTMLAudioElement) {
+        element.autoplay = true;
+
         this.remoteVideoElement.muted = true;
         this.remoteAudioElement = element;
         this.remoteAudioElement.muted = false;
-        this.tryPlayRemoteAudioStream();
+
+        this.queueMediaOperation(MediaQueueId.REMOTE_AUDIO, async () => {
+            // if audioOutput is non-default:
+            try {
+                if (audioOutput) await element.setSinkId(audioOutput);
+            } catch (e) {
+                logger.warn("Couldn't set requested audio output device: using default", e);
+            }
+
+            element.srcObject = this.remoteStream;
+            return element.play();
+        });
     }
 
     /**
      * Configure this call from an invite event. Used by MatrixClient.
      * @param {MatrixEvent} event The m.call.invite event
      */
-    initWithInvite(event : any) {
+    async initWithInvite(event : any) {
```

Comment:
can this get a type?

Reply:
do we have a type for events in the js-sdk yet?

Reply:
No but you can use `MatrixEvent` (imported) and it'll work magically as soon as that gets Typescripted, helps prevent future issues from cropping up during MatrixEvent being TSified